src/pyiterator.jl

```diff
@@ -29,16 +30,59 @@ end
 
     Base.done(po::PyObject, s) = ispynull(s[1])
 else
-    function Base.iterate(po::PyObject, s=_start(po))
+    """
+        PyIterator{T}(pyobject)
+
+    Wrap `pyobject::PyObject` into an iterator, that produces items of type `T`. To be more precise `convert(T, item)` is applied in each iteration. This can be useful to avoid automatic conversion of items into corresponding julia types.
+    ```jldoctest
+    julia> using PyCall
+
+    julia> l = PyObject([PyObject(1), PyObject(2)])
+    PyObject [1, 2]
+    
+    julia> piter = PyCall.PyIterator{PyAny}(l)
+    PyCall.PyIterator{PyAny}(PyObject [1, 2])
+    
+    julia> collect(piter)
+    2-element Array{Any,1}:
+     1
+     2
+    
+    julia> piter = PyCall.PyIterator{PyObject}(l)
+    PyCall.PyIterator{PyObject}(PyObject [1, 2])
```

Comment:
Probably this should be the default: there should be a `PyIterator(o::PyObject) = PyIterator{PyObject}(o)` constructor.

Reply:
Good idea, I added it.